client/clientutil/object.go

```diff
@@ -202,19 +203,151 @@ func CreateObjectWithClient(ctx context.Context, c *client.Client, prm CreateObj
 // inherits behavior of CreateObjectWithClient.
 func CreateObject(ctx context.Context, endpoint string, prm CreateObjectPrm) error {
-	var prmInit client.PrmInit
-	prmInit.ResolveNeoFSFailures()
+	c, err := createClient(endpoint)
+	if err != nil {
+		return err
+	}
+
+	return CreateObjectWithClient(ctx, c, prm)
+}
+
+// ReadObjectPrm groups parameters of ReadObject operation.
+type ReadObjectPrm struct {
+	// Target NeoFS container.
+	Container cid.ID
+
+	// Object's reference to read from the Container.
+	Object oid.ID
 
-	var c client.Client
-	c.Init(prmInit)
+	payloadWriter io.Writer
+}
 
-	var prmDial client.PrmDial
-	prmDial.SetServerURI(endpoint)
+// WritePayloadTo sets optional io.Writer to write payload of the request
+// object to.
+func (x *ReadObjectPrm) WritePayloadTo(w io.Writer) {
+	x.payloadWriter = w
+}
 
-	err := c.Dial(prmDial)
+// ReadObjectWithClient reads object from the NeoFS network using the given
+// client.
+//
+// The object is read from the container referenced by ReadObjectPrm.Container
+// which MUST be explicitly set. Exact object is referenced by
+// ReadObjectPrm.Object which MUST be explicitly set.
+//
+// Container SHOULD be public-read. ReadObjectWithClient uses random private key
+// for communication over the NeoFS protocol. This is also suitable in the absence
+// of a specific key.
+//
+// By default, ReadObjectWithClient reads but discards the object data. It can
+// be used to check object availability. To explicitly specify payload target,
+// use ReadObjectPrm.WritePayloadTo method.
+//
+// Client connection MUST be opened in advance, see Dial method for details.
+// Network communication is carried out within a given context, so it MUST NOT
+// be nil.
+//
+// See also ReadObject.
+func ReadObjectWithClient(ctx context.Context, c *client.Client, prm ReadObjectPrm) error {
+	// initialize object stream
+	var prmGet client.PrmObjectGet
+	prmGet.FromContainer(prm.Container)
+	prmGet.ByID(prm.Object)
+	prmGet.UseKey(signerDefault)
+
+	streamObj, err := c.ObjectGetInit(ctx, prmGet)
+	if err != nil {
+		return fmt.Errorf("init object writing on client: %w", err)
+	}
+
+	// read and discard header
+	if streamObj.ReadHeader(new(object.Object)) {
+		if prm.payloadWriter == nil {
+			// discard payload by default
+			prm.payloadWriter = io.Discard
+		}
+
+		// copy payload to the destination
+		_, err = io.Copy(prm.payloadWriter, streamObj)
+		if err != nil {
+			return err
+		}
+	}
+
+	_, err = streamObj.Close()
+	if err != nil {
+		return fmt.Errorf("read object: %w", err)
+	}
+
+	return nil
+}
+
+// ReadObject reads object from the NeoFS network through the given endpoint.
+//
+// ReadObject is well suited for one-time data reading. To read multiple
+// objects using the same endpoint, use ReadObjectWithClient. ReadObject
+// inherits behavior of ReadObjectWithClient.
+func ReadObject(ctx context.Context, endpoint string, prm ReadObjectPrm) error {
```

Comment:
Maybe these could be some `quick` API, while the other are more like Client (or upper-level thing) methods.